### PR TITLE
[AI Widget] file support

### PR DESCRIPTION
**Pull request by bang9**
## For Internal Contributors

* Follow the [Scaled Trunk-Based Development workflow](https://trunkbaseddevelopment.com/)
* Branch naming format: `{type}/TICKET_ID/description`
  * Type: `feat` / `fix` / `chore` / `doc` / `release`
* Receive PR review approvals
* Rebase your branch with the main branch and wait for CI to pass
* Squash merge your commit
  * Use imperative language in the title and description
  * Follow the provided template for PR description and squashing

### Template
```
// PR title (Required)
[type]: A short description of the changes in imperative language.

// PR description (Optional)
Add a brief description of the changes in this PR. Bullet points are also fine.

// Footer (Recommended)
Fixes [<TICKET_ID>](https://sendbird.atlassian.net/browse/<TICKET_ID>)

// Changelogs (Recommended)
// Add (internal) at the end of each changelog if internal.
### Changelogs

// Co-authors
// Add this if you pair programmed or they made significant contributions to the ideas in the code and you want to thank them.
Co-authored-by: Name name@example.com, Name2 name@example.com

```

### Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If unsure, ask the members.
This is a reminder of what we look for before merging your code.

- [ ] **All tests pass locally with my changes**
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **Public components / utils / props are appropriately exported**
- [ ] I have added necessary documentation (if appropriate)

## External Contributions

This project is not yet set up to accept pull requests from external contributors.

If you have a pull request that you believe should be accepted, please contact
the Developer Relations team <developer-advocates@sendbird.com> with details
and we'll evaluate if we can set up a [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement) to allow for the contribution.

### CODE REVIEW

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx [hunk 1]
```diff
@@ -96,7 +96,7 @@ export const useHandleUploadFiles = (
       if (sendingFiles.length === 1) {
         logger.info('Channel|useHandleUploadFiles: sending one file.');
         const [file] = sendingFiles;
-        return sendFileMessage({ file, parentMessageId: quoteMessage?.messageId });
+        return sendFileMessage({ file, parentMessageId: quoteMessage?.messageId, ...params });
```
Comment (liamcho) [inline src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx:99]: Let's move `...params` to the most front.

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx [hunk 2]
```diff
@@ -114,23 +114,24 @@ export const useHandleUploadFiles = (
         return otherFiles.reduce(
           (previousPromise: Promise<MultipleFilesMessage | FileMessage | void>, item: File) => {
             return previousPromise.then(() => {
-              return sendFileMessage({ file: item, parentMessageId: quoteMessage?.messageId });
+              return sendFileMessage({ file: item, parentMessageId: quoteMessage?.messageId, ...params });
```
Comment (liamcho) [inline src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx:117]: Let's move `...params` to the most front.

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx [hunk 3]
```diff
@@ -114,23 +114,24 @@ export const useHandleUploadFiles = (
...
             });
           },
           (() => {
             if (imageFiles.length === 0) {
               return Promise.resolve();
             } else if (imageFiles.length === 1) {
-              return sendFileMessage({ file: imageFiles[0] });
+              return sendFileMessage({ file: imageFiles[0], ...params });
```
Comment (liamcho) [inline src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx:124]: Let's move `...params` to the most front.

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx [hunk 4]
```diff
@@ -114,23 +114,24 @@ export const useHandleUploadFiles = (
...
-              return sendMultipleFilesMessage({
+              return sendMultipleFilesMessage?.({
                 fileInfoList: imageFiles.map((file) => ({
                   file,
                   fileName: file.name,
                   fileSize: file.size,
                   mimeType: file.type,
                 })),
                 parentMessageId: quoteMessage?.messageId,
+                ...params,
```
Comment (liamcho) [inline src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx:134]: Let's move `...params` to the most front.

File: src/ui/MessageInput/hooks/useFileUploadButton.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+  const addFiles = useCallback(
+    (newFiles: File[]) => {
+      const filtered = newFiles.filter((it) => (accept ? accept.includes(it.type) : true));
+      if (filtered.length === 0) return;
+      if (!multiple && filtered.length > 1) {
+        onLimitReached?.();
+        return;
+      }
+      if (!multiple && filtered.length > 0 && files.length > 0) {
```
Comment (liamcho) [inline src/ui/MessageInput/hooks/useFileUploadButton.tsx:28]: Does this logic allow adding only one item?

Comment (bang9) [reply src/ui/MessageInput/hooks/useFileUploadButton.tsx:28]: 다음과 같은 조건입니다.
- 멀티플이 아닌데, 추가하려는 파일이 2개 이상일 때
- 멀티플이 아닌데, 기존에 추가된 파일이 1개 이상이고 추가하려는 파일이 1개 이상일 때

Comment (bang9) [reply src/ui/MessageInput/hooks/useFileUploadButton.tsx:28]: 현재 멀티플은 지원되지 않기 때문에 하드코딩되어 있습니다.

File: src/ui/MessageInput/hooks/useFileUploadButton.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+interface Params {
+  disabled?: boolean;
+  accept?: string;
+  multiple?: boolean;
+  onLimitReached?: () => void;
+}
+
+export function useFileUploadButton({ accept, multiple, disabled, onLimitReached }: Params) {
+  const [files, setFiles] = useState<File[]>([]);
+  const [filePreview, setFilePreview] = useState<{ url: string; type: string; name: string }[]>([]);
```
Comment (liamcho) [inline src/ui/MessageInput/hooks/useFileUploadButton.tsx:15]: how about combining into single state?

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx (hunk 1 above)
Comment (bang9) [reply src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx]: addressed in https://github.com/sendbird/sendbird-uikit-react/pull/1218/commits/474a8e55880595639d5b51f3fa6eb1340e2bd8f8

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx (hunk 2 above)
Comment (bang9) [reply src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx]: addressed in https://github.com/sendbird/sendbird-uikit-react/pull/1218/commits/474a8e55880595639d5b51f3fa6eb1340e2bd8f8

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx (hunk 4 above)
Comment (bang9) [reply src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx]: addressed in https://github.com/sendbird/sendbird-uikit-react/pull/1218/commits/474a8e55880595639d5b51f3fa6eb1340e2bd8f8

File: src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx (hunk 3 above)
Comment (bang9) [reply src/modules/GroupChannel/components/MessageInputWrapper/useHandleUploadFiles.tsx]: addressed in https://github.com/sendbird/sendbird-uikit-react/pull/1218/commits/474a8e55880595639d5b51f3fa6eb1340e2bd8f8

File: src/ui/MessageInput/hooks/useFileUploadButton.tsx (hunk 6 above)
Comment (bang9) [reply src/ui/MessageInput/hooks/useFileUploadButton.tsx:15]: good idea~ let me try

File: src/ui/MessageInput/hooks/useFileUploadButton.tsx (hunk 5 above)
Comment (liamcho) [reply src/ui/MessageInput/hooks/useFileUploadButton.tsx:28]: 설명 감사합니다 이해했습니다

File: src/ui/MessageInput/hooks/useFileUploadButton.tsx (hunk 6 above)
Comment (bang9) [reply src/ui/MessageInput/hooks/useFileUploadButton.tsx]: addressed in https://github.com/sendbird/sendbird-uikit-react/pull/1218/commits/a57fb645a6e6a02323306de19438c86c2f258e97

**Review by liamcho — APPROVED** [review]
LGTM

**Pull request merged**